apps/emqx_durable_storage/src/emqx_ds_new_streams.erl

```diff
@@ -0,0 +1,405 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+%% @doc This module is used for notifying processes about new streams.
+%% It's not a replacement for `emqx_ds:get_streams' function, it's
+%% only meant to optimize its usage.
```

Comment:
The relation of the API that this module provides to the DS API may be still unclear: is it available for all DS'es? Which DS'es should implement it? How is it possible for a DS consumer understand whether it is possible to use it?

Reply:
Right now the idea is that any DS backend/layout can use this module at its own discretion.